### PR TITLE
Milestone 2: Components and Redux store

### PR DESCRIPTION
**Pull request by SotoArmando**
Your project structure and basic dependencies are now ready. Great! Now, it is time to initialize the Redux actions, reducers and store. Additionally, we'll also define empty React components. Remember to separate presentation and container components.

### CODE REVIEW

File: src/Components/booksform.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+const booksform = () => {
+  const categories = ['Action', 'Biography', 'History', 'Horror', 'Kids', 'Learning', 'Sci-Fi'];
+  return (
+    <div className="App">
+      {categories}
+    </div>
+  );
+};
+
+export default booksform;
```
Comment (shloch) [inline src/Components/booksform.js]: - `booksform.js should be placed in folder `src/containers`
- should have a state with `title` and `category`
- it should renders form with title input, categories select box and submit [[screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/bookstore/milestone2/Form/RenderForm.png)]

File: src/Components/Bookslist.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  return (
+    <div className="Menu">
+      <Msgtitle />
+      <Booksform />
+      {bookstore.map(ele => <Book key={ele.id} book={ele} />)}
+    </div>
+  );
+}
+
+const mapStateToProps = state => ({ bookstore: state.book });
```
Comment (MauricioRobayo) [inline src/Components/Bookslist.js:26]: Please move components that connect to the Redux store to the containers folder, as you did with booksform.js.

File: src/Containers/msgtitle.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+function Msgtitle() {
+  return (
+    <div className="Msgtitle">
+      <span className="title fw-700">All your steps, One steps place</span>
+      <span className="desc">Keep track over all your books just by adding those with just one key</span>
+    </div>
+  );
+}
+
+export default Msgtitle;
```
Comment (MauricioRobayo) [inline src/Containers/msgtitle.js:12]: This seems like a presentational component, should be on components directory.

Comment (SotoArmando) [reply src/Containers/msgtitle.js:12]: Thanks for the notes I think I got to save them xD @MauricioRobayo. they will be usefull 🚀.

File: src/Components/Ennumerated_Components/Book.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+          %
+        </span>
+      </div>
+    </div>
+  );
+}
+
+const mapDispatchToProps = dispatch => ({
+  // dispatching plain actions
+  // CREATE_BOOK: (book) => dispatch({ type: 'CREATE_BOOK', book }),
```
Comment (bellom) [inline src/Components/Ennumerated_Components/Book.js:34]: It’s better to delete commented code. If you need to review a previous version of your code, use git history. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/no_inline_comments.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))

File: src/Components/Ennumerated_Components/Book.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
```
Comment (bellom) [inline src/Components/Ennumerated_Components/Book.js:3]: Job well-done @SotoArmando, 👍 
According to the previous review, you were supposed to move this to the container folder right?

**I quote from @MauricioRobayo comments** 
`One of the main purposes of this milestone is learning about presentational components and container components.`

`Container components will manage the state (in this case the ones that will connect to the Redux store). Presentational components are stateless`

**In other words;** In React file structure, any component that is connecting to the redux store should be in the `container folder`.

[Helpful Article](https://krasimirtsonev.com/blog/article/react-js-presentational-container-components)
